pkg/api/api_test.go

```diff
@@ -242,52 +243,48 @@ func TestPostageHeaderError(t *testing.T) {
 	var (
 		mockStorer     = mock.NewStorer()
 		mockStatestore = statestore.NewStateStore()
-		logger         = logging.New(ioutil.Discard, 0)
+		logger         = logging.New(ioutil.Discard, 5)
 		client, _, _   = newTestServer(t, testServerOptions{
 			Storer: mockStorer,
 			Tags:   tags.NewTags(mockStatestore, logger),
 			Logger: logger,
 		})
-	)
 
-	for _, tc := range []struct {
-		name     string
-		endpoint string
-		batchId  []byte
-		expErr   bool
-	}{
-		{
-			name:     "bytes",
-			endpoint: "/bytes",
-			batchId:  []byte{0},
-			expErr:   true,
-		},
-		{
-			name:     "bytes",
-			endpoint: "/bytes",
-			batchId:  []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, //32 bytes - ok
-		},
-		{
-			name:     "bytes",
-			endpoint: "/bytes",
-			batchId:  []byte{}, // empty batch id falls back to the default zero batch id, so expect 200 OK. This will change once we do not fall back to a default value
-		},
-		{
-			name:     "bzz",
-			endpoint: "/bzz",
-			batchId:  []byte{0},
-			expErr:   true,
-		},
-	} {
+		batchInvalid = []byte{0}
+		batchOk      = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0} //32 bytes - ok
+		batchEmpty   = []byte{}
 
-		t.Run(tc.name, func(t *testing.T) {
-			hexbatch := hex.EncodeToString(tc.batchId)
+		endpoints = []string{
+			"bytes", "bzz", "chunks",
+		}
+	)
+
+	for _, endpoint := range endpoints {
+		t.Run(endpoint+": empty batch", func(t *testing.T) {
+			hexbatch := hex.EncodeToString(batchEmpty)
 			expCode := http.StatusOK
-			if tc.expErr {
-				expCode = http.StatusBadRequest
-			}
-			jsonhttptest.Request(t, client, http.MethodPost, tc.endpoint, expCode,
+			jsonhttptest.Request(t, client, http.MethodPost, "/"+endpoint, expCode,
+				jsonhttptest.WithRequestHeader(api.SwarmPostageBatchIdHeader, hexbatch),
+				jsonhttptest.WithRequestHeader(api.ContentTypeHeader, "application/octet-stream"),
+				jsonhttptest.WithRequestBody(bytes.NewReader([]byte{0, 0, 0, 0, 0, 0, 0, 0})),
```

Comment:
r: if you want to short your byte slice literals in this line and below, you can define them as `[]byte{0: 0, 7: 0}`